c/src/ml-api-service-offloading.c

```diff
@@ -485,6 +488,42 @@ _mlrs_process_service_offloading (nns_edge_data_h data_h, void *user_data)
       event_type = ML_SERVICE_EVENT_REPLY;
       break;
     }
+    case ML_SERVICE_OFFLOADING_TYPE_LAUNCH:
+    {
+      ml_service_h service_h = NULL;
+
+      /**
+       * @todo Check privilege and availability here.
+       */
+
+      service_h =
+          g_hash_table_lookup (offloading_s->service_table, service_key);
+      if (service_h) {
+        _ml_logi ("The registered service as key %s is already launched.",
+            service_key);
+        break;
+      }
+
+      ret = ml_service_pipeline_launch (service_key, &service_h);
+      if (ret != ML_ERROR_NONE) {
+        _ml_error_report
+            ("Failed to launch the registered pipeline. service key: %s",
+            service_key);
+        goto done;
+      }
+      ret = ml_service_start (service_h);
+      if (ret != ML_ERROR_NONE) {
+        _ml_error_report
```

Comment:
mem leak, need releasing service_h if failed to start pipeline.

Reply:
Thanks, Changed to release the handle.